CONTRIBUTING.md

```diff
@@ -95,123 +97,272 @@ cd docs
-    test_submit_ok (tests.hooks.test_druid_hook.TestDruidHook) ... ok
-    test_submit_timeout (tests.hooks.test_druid_hook.TestDruidHook) ... ok
-    test_submit_unknown_response (tests.hooks.test_druid_hook.TestDruidHook) ... ok
+Once initialization is done, you should select the virtualenv you initialized as the
+project's default virtualenv in your IDE and run tests efficiently.
 
-    ----------------------------------------------------------------------
-    Ran 8 tests in 3.036s
+After setting it up - you can use the usual "Run Test" option of the IDE and have
+the autocomplete and documentation support from IDE as well as you can
+debug and view the sources of Airflow - which is very helpful during
+development.
 
-    OK
-  ```
+### Running tests
+
+To run tests locally, once you activate virtualenv you should be able to simply run
+``run-tests`` at will. Note that if you want to pass extra parameters to nose
+you should do it after '--'
+
+For example, in order to just execute the "core" unit tests, run the following:
 
-  The Airflow code is mounted inside of the Docker container, so if you change something using your favorite IDE, you can directly test it in the container.
+```bash
+run-tests tests.core:CoreTest -- -s --logging-level=DEBUG
+```
+or a single test method:
 
-3. Using [Docker Compose](https://docs.docker.com/compose/) and Airflow's CI scripts
+```bash
+run-tests tests.core:CoreTest.test_check_operators -- -s --logging-level=DEBUG
+```
 
-  Start a docker container through Compose for development to avoid installing the packages directly on your system. The following will give you a shell inside a container, run all required service containers (MySQL, PostgresSQL, krb5 and so on) and install all the dependencies:
+### Running tests directly from the IDE
 
-  ```bash
-  docker-compose -f scripts/ci/docker-compose.yml run airflow-testing bash
-  # From the container
-  export TOX_ENV=py35-backend_mysql-env_docker
-  /app/scripts/ci/run-ci.sh
-  ```
+Once you configure your tests to use the virtualenv you created. running tests
+from IDE is as simple as:
 
-  If you wish to run individual tests inside of Docker environment you can do as follows:
+![Run unittests](images/run_unittests.png)
 
-  ```bash
-  # From the container (with your desired environment) with druid hook
-  export TOX_ENV=py35-backend_mysql-env_docker
-  /app/scripts/ci/run-ci.sh -- tests/hooks/test_druid_hook.py
-  ```
+Note that while most of the tests are typical "unit" tests that do not
+require external components, there are a number of tests that are more of
+"integration" ot even "system" tests (depending on the convention you use).
+Those tests interact with external components. For those tests
+you need to run complete Docker Compose - based
+[Integration test development environment](#integration-test-development-environment).
 
+## Docker container development environment
```

Comment:
Have you figured out how to do docker-compose-based testing within an IDE? I was trying to do so within pycharm and found it pretty unintuitive. This would be "best of both worlds" if possible.